pkg/helpers/newapp/app/pipeline.go

```diff
@@ -396,6 +423,26 @@ func AddServices(objects Objects, firstPortOnly bool) Objects {
 			if svc != nil {
 				svcs = append(svcs, svc)
 			}
+		case *kappsv1.Deployment:
+			svc := addService(t.Spec.Template.Spec.Containers, t.ObjectMeta, t.Spec.Template.Labels, firstPortOnly)
+			if svc != nil {
+				svcs = append(svcs, svc)
+			}
+		case *extensionsv1beta1.Deployment:
```

Comment:
Didn't we remove this in 4.4? Or are we keeping logic here for backwards compatibility?

Reply:
Backwards compatibility, that is correct.